src/protocol/bolt/src/main/java/org/apache/jmeter/protocol/bolt/sampler/BoltSampler.java

```diff
@@ -0,0 +1,174 @@
+
+    @Override
+    public SampleResult sample(Entry e) {
+        SampleResult res = new SampleResult();
+        res.setSampleLabel(getName());
+        res.setSamplerData(request());
+        res.setDataType(SampleResult.TEXT);
+        res.setContentType("text/plain"); // $NON-NLS-1$
+        res.setDataEncoding(StandardCharsets.UTF_8.name());
+
+        Map<String, Object> params;
+        try {
+            params = getParamsAsMap();
+        } catch (IOException ex) {
+            return handleException(res, ex);
+        }
+
+        // Assume we will be successful
+        res.setSuccessful(true);
+        res.setResponseMessageOK();
+        res.setResponseCodeOK();
+
+        res.sampleStart();
+
+        try {
+            res.setResponseHeaders("Cypher request: " + getCypher());
+            res.setResponseData(execute(BoltConnectionElement.getDriver(), getCypher(), params), StandardCharsets.UTF_8.name());
+        } catch (Exception ex) {
+            res = handleException(res, ex);
+        } finally {
+            res.sampleEnd();
+        }
+        return res;
+    }
+
+    /**
+     * @see org.apache.jmeter.samplers.AbstractSampler#applies(org.apache.jmeter.config.ConfigTestElement)
+     */
+    @Override
+    public boolean applies(ConfigTestElement configElement) {
+        String guiClass = configElement.getProperty(TestElement.GUI_CLASS).getStringValue();
+        return APPLICABLE_CONFIG_CLASSES.contains(guiClass);
+    }
+
+    private String execute(Driver driver, String cypher, Map<String, Object> params) {
+        try (Session session = driver.session()) {
+            StatementResult statementResult = session.run(cypher, params);
+            return response(statementResult);
+        }
+    }
+
+    private SampleResult handleException(SampleResult res, Exception ex) {
+        res.setResponseMessage(ex.toString());
+        if (ex instanceof Neo4jException) {
+            res.setResponseCode(((Neo4jException)ex).code());
+        } else {
+            res.setResponseCode("500");
+        }
+        res.setResponseData(ObjectUtils.defaultIfNull(ex.getMessage(), "NO MESSAGE").getBytes());
+        res.setSuccessful(false);
+        return res;
+    }
+
+    private Map<String, Object> getParamsAsMap() throws IOException {
+        if (getParams() != null && getParams().length() > 0) {
+            return objectMapper.readValue(getParams());
+        } else {
+            return Collections.emptyMap();
+        }
+    }
+
+    private String request() {
+        StringBuilder request = new StringBuilder();
+        request.append("Query: \n")
+                .append(getCypher())
+                .append("\n")
+                .append("Parameters: \n")
+                .append(getParams());
+        return request.toString();
+    }
+
+    private String response(StatementResult result) {
+        StringBuilder response = new StringBuilder();
```

Comment:
Is there a way to compute latency ?

Reply:
AFAIK there is not such info in the query results

Reply:
But can you still measure it as `timestampBeforeResultProcessing - timestampBeforeQueryStart` ?

Reply:
when the `session.run` returns, the driver executed the query, but also started streaming some results. so the time we can compute client side corresponds to more than the actual response from the database. 
but I guess there's no harm into adding it. will do

Reply:
@nmervaillie, if it's not meaningful, I don't think it's useful to add it. It will incur more issues if people analyze deeply results.

Reply:
It will help for the analysis, especially if documentation would clarify how results are represented.
For instance, how "number of rows", "number of columns", "number of bytes", "time to first row" and so on are mapped to the results in JMeter.

Reply:
I guess you have to come to an agreement guys ;-)

Reply:
You're the Neo4j specialist, so if you think computing latency this way is useful go ahead.